overcast/runner/__init__.py

```diff
@@ -413,16 +457,35 @@ def provision_step(self, details):
         for base_node_name, node_info in stack['nodes'].items():
             def _create_node(base_name):
                 node_name = self.add_prefix(base_name)
-                self.nodes[base_name] = self.create_node(node_name, node_info,
-                                                         keypair=keypair_name,
-                                                         userdata=userdata)
+                self.nodes[base_name] = node(node_name, node_info,
```

Comment:
self will be passed to Node - so mappings can be removed from here as it is passed to dr, so he have that data,. May be it would be something like this

self.nodes[base_name] = Node(node_name, node_info, keypair=keypair_name, userdata=userdata,dr=self)